mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -0,0 +1,479 @@
+#!/usr/bin/env python3
+
+
+import csv
+import getopt
+import os
+import subprocess
+import sys
+import math
+from prettytable import PrettyTable
+from prettytable import from_csv 
+from datetime import date
+
+# global variables.
+mlirBuildDir = './bin'
+mlirMIOpenDriver = 'mlir-miopen-driver'
+mlirROCmRunner = 'mlir-rocm-runner'
+rocprof = '/opt/rocm/bin/rocprof'
+MIOpenDriver = os.path.expanduser('~/MIOpen/build/bin/MIOpenDriver')
+benchmarkingResultFileName = 'results.stats.csv'
+configurationFileName ='../mlir/utils/jenkins/miopen-tests/resnet50-miopen-configs'
+roundDigits = 2
+
+Directions = ['-F 1', '-F 4']
+DataTypes = ['conv', 'convfp16']
+Layouts = ['NCHW']
+#Layouts = ['NHWC', 'NCHW']
+
+# rocprof crashed with MIOpenDriver for these configurations on gfx908
+DisabledConfigs = ['-n 256 -c 256 -H 56 -W 56 -k 128 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1',
+                   '-n 256 -c 256 -H 56 -W 56 -k 64 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1',
+                   '-n 256 -c 64 -H 56 -W 56 -k 256 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1']
+
+# utility functions.
+def getConfigurations(fileName):
+    xdlops = False;
+    r = subprocess.run(f'/opt/rocm/bin/rocm_agent_enumerator -t GPU|grep gfx908', shell=True)
+    if r.returncode == 0:
+        xdlops = True
+
+    configFile = open(fileName, 'r')
+    Lines = configFile.readlines()
+    configs = [];
+    for direction in Directions: 
```

Comment:
This could be replaced with a call to [itertools.product](https://docs.python.org/3/library/itertools.html#itertools.product).